### PR TITLE
feat(uniswapx-sdk): Update Structs & ABI functionality for Dutch V3 Gas Adjustment Feature

### PR DESCRIPTION
**Pull request by alanhwu**
## Description

Structural changes for DutchV3 to support new gas adjustment feature

This unblocks us to now write E2E integration tests

## How Has This Been Tested?

Tests have been updated to check and use this field. There is no formal checking of logical bounds of the feature, which should be done when we implement support for this feature to be used.

### CODE REVIEW

File: sdks/uniswapx-sdk/src/builder/V3DutchOrderBuilder.test.ts [hunk 1]
```diff
@@ -364,13 +406,51 @@ describe("V3DutchOrderBuilder", () => {
           ).toThrow("Invariant failed: nonce not set");
     });
 
+    it("Throw if startingBaseFee not set", () => {
```
Comment (codyborn) [inline sdks/uniswapx-sdk/src/builder/V3DutchOrderBuilder.test.ts:409]: Maybe add a test for when `startingBaseFee` and  `adjustmentPerGweiBaseFee` are non-zero (e.g. `1e9`)

**Review by codyborn — APPROVED** [review]
Looks good! Left a minor comment.

**Review by zhongeric — APPROVED** [review]

File: sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts [hunk 2]
```diff
@@ -303,17 +311,436 @@ describe.only("DutchV3Order", () => {
...
+                    relativeAmounts: [BigInt(0)],
+                },
+                maxAmount: SMALL_AMOUNT,
+                adjustmentPerGweiBaseFee: BigNumber.from(0),
+            })
+            .output({
+                token: tokenOut.address,
+                startAmount: SMALL_AMOUNT,
+                curve: {
+                    relativeBlocks: [1],
```
Comment (codyborn) [inline sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts:340]: No decay can also be achieved with two empty arrays and I think in practice that's preferable for gas savings.

File: sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts [hunk 3]
```diff
@@ -303,17 +311,436 @@ describe.only("DutchV3Order", () => {
...
+
+        expect((await tokenOut.balanceOf(swapperAddress)).toString()).to.equal(
+            swapperTokenOutBalanceBefore.add(SMALL_AMOUNT)
+        );
+        expect((await tokenOut.balanceOf(fillerAddress)).toString()).to.equal(
+            fillerTokenOutBalanceBefore.sub(SMALL_AMOUNT)
+        );
+    });
+
+    it("executes a serialized order with decay", async () => {
```
Comment (codyborn) [inline sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts:479]: Could you also add a test for a multi-point decay with different values?

Comment (alanhwu) [reply sdks/uniswapx-sdk/integration/test/V3DutchOrder.spec.ts:479]: added!

**Review by codyborn — APPROVED** [review]
Looks great! Nice work
